File: Medium-Array/Twosum.java
// WAP to find the two elements in array that add up to a given sum is equal to target.

// This program checks whether any two numbers in an array sum up to a given target value.
import java.util.Scanner;

class Twosum
{
    // Method to check if there exists a pair of elements in the array whose sum equals the target.
    boolean twoSum(int arr[], int target) 
    {
        // Loop through each element in the array
        for(int i=0;i<arr.length;i++)
        {
            // Loop through the rest of the array to find a pair
            for(int j=i+1;j<arr.length;j++)
            {
                // Check if the sum of the current pair equals the target
                if(arr[i]+arr[j]==target)
                {
                    return true;
                }
            }
        }
        return false;
    }
    // Main method to take user input and invoke the twoSum method
    public static void main(String[] args)
    {
        // Create a Scanner object to take input from the user
       Scanner sc=new Scanner(System.in);
       System.out.println("Size of Array");
       // Read the size of the array
       int n=sc.nextInt();
       int arr[]=new int[n];
       System.out.println("Enter the elements of array");
       // Read the elements of the array
       for(int i=0;i<n;i++)
       {
        arr[i]=sc.nextInt();
       }
       System.out.println("Enter the target sum");
       // Read the target sum
       int target=sc.nextInt();
       // Create an instance of Twosum class
       Twosum ts=new Twosum();
       // Check if there are two elements that add up to the target
       if(ts.twoSum(arr,target))
       {
        System.out.println("Two elements in array that add up to a given sum is equal to target");
       }
       else
       {
        System.out.println("No two elements in array that add up to a given sum is equal to target");
        }
    }
}

File: Algorithm/twosum.py
# two sum with target sum
# using two pointers approach
def two_sum(nums, target):
    nums.sort()
    left, right = 0, len(nums) - 1

    while left < right:
        current_sum = nums[left] + nums[right]

        if current_sum == target:
            return [nums[left], nums[right]]
        elif current_sum < target:
            left += 1
        else:
            right -= 1

    return []

# Example usage
num=int(input("Enter the number of elements in the array: "))
nums = []
print("Enter the elements of the array:")
for _ in range(num):
    nums.append(int(input()))
target = int(input("Enter the target sum: "))
result = two_sum(nums, target)
if result:
    print(f"Two numbers that add up to {target} are: {result[0]} and {result[1]}")
else:
    print(f"No two numbers add up to {target}.")
